internal/hcs/system.go

```diff
@@ -541,26 +540,64 @@ func (computeSystem *System) CreateProcess(c interface{}) (_ cow.Process, err er
 	})
 	events := processHcsResult(resultp)
 	if err != nil {
-		return nil, makeSystemError(computeSystem, "CreateProcess", configuration, err, events)
+		return nil, 0, makeSystemError(computeSystem, "CreateProcess", configuration, err, events)
 	}
 
 	logrus.WithFields(computeSystem.logctx).
 		WithField(logfields.ProcessID, processInfo.ProcessId).
 		Debug("HCS ComputeSystem CreateProcess PID")
 
+	return &processInfo, processHandle, nil
+}
+
+// CreateProcessNoStdio launches a new process within the computeSystem. The
+// Stdio handles are not cached on the process struct.
+func (computeSystem *System) CreateProcessNoStdio(c interface{}) (_ cow.Process, err error) {
+	processInfo, processHandle, err := computeSystem.createProcess(c)
```

Comment:
Cant we do this via a type check on `c` to be `*ProcessConfig` and from that we know that its `v1`?

Reply:
That introduces an import cycle between `hcsshim` and `hcsshim/internal/hcs`. I think it's better to be explicit anyways, since we control the caller.

Reply:
It's not really about v1 vs v2 anyway, it's really about whether this instance of the Process object should take ownership of the stdio handles from HCS.